core/src/main/java/tachyon/util/NettyUtils.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.util;
+
+import java.util.concurrent.ThreadFactory;
+
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.ServerChannel;
+import io.netty.channel.epoll.EpollEventLoopGroup;
+import io.netty.channel.epoll.EpollServerSocketChannel;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.nio.NioServerSocketChannel;
+
+import tachyon.worker.netty.ChannelType;
+
+public final class NettyUtils {
+
+  private NettyUtils() {}
+
+  /**
+   * Creates a Netty EventLoopGroup based on the IOMode.
+   *
+   * @param type Selector for which form of low-level IO we should use
+   * @param numThreads
+   * @param threadPrefix name pattern for each thread. should contain '%d' to distinguish between
+   *        threads.
+   */
+  public static EventLoopGroup createEventLoop(ChannelType type, int numThreads,
+      String threadPrefix) {
+    ThreadFactory threadFactory = ThreadFactoryUtils.build(threadPrefix);
+
+    switch (type) {
+      case NIO:
+        return new NioEventLoopGroup(numThreads, threadFactory);
+      case EPOLL:
+        return new EpollEventLoopGroup(numThreads, threadFactory);
+      default:
+        throw new IllegalArgumentException("Unknown io mode: " + type);

```

Comment:
I think you forgot to change the error message when you copied this code from Spark.